### PR TITLE
Update moltres_xs.py to be compatible with OpenMC 0.13.2

### LINKED ISSUE #205: moltres_xs.py maybe not compatiable with the newest version of OpenMC
An error appeared when I used moltres_xs.py to generate group constants. I think maybe `v0.13.0` OpenMC doesn't use `legendre` any more and I think it uses `mean` instead.
error

### PR DESCRIPTION
**Pull request by smpark7**
Fixes #205 

The following changes from OpenMC versions 0.12.2 to 0.13.2 affected `moltres_xs.py`:
- Renaming the `groups` parameter for `MGXS` group constant classes to `energy_groups`
  - openmc-dev/openmc#2147
- Removing the `legendre` filter in the diffusion coefficient group constants
  - openmc-dev/openmc#1909

### CODE REVIEW

**Comment by smpark7** [comment]
I just realized that this breaks backward compatibility with pre-0.13.2 OpenMC-generated files.

**Comment by munkm** [comment]
> I just realized that this breaks backward compatibility with pre-0.13.2 OpenMC-generated files.

I was going to ask that! Ok, so maybe it would be helpful to add a warning or helpful error to users trying to load files that are that old? What do you think?

File: tutorial/godiva/godiva_openmc_1200.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 geom.export_to_xml()
 
 settings = openmc.Settings()
-settings.batches = 200
+settings.batches = 100
```
Comment (munkm) [inline tutorial/godiva/godiva_openmc_1200.py:24]: Is this change to save time?

Comment (smpark7) [reply tutorial/godiva/godiva_openmc_1200.py:24]: I'm not sure why the simulation at 900K used 100 batches while 1200K used 200 batches. It seems like a mistake so I wanted to standardize both to 100 batches.

File: python/moltres_xs.py [hunk 2]
```diff
@@ -300,10 +299,10 @@ def generate_openmc_tallies_xml(
                 name=str(id) + "_beta",
             )
             domain_dict[id]["chi"] = mgxs.Chi(
-                domain=domain, groups=groups, name=str(id) + "_chi"
+                domain=domain, energy_groups=groups, name=str(id) + "_chi"
```
Comment (munkm) [inline python/moltres_xs.py:302]: I think because we're switching the kwarg it also means we aren't supporting older openmc versions.

**Comment by munkm** [comment]
I think it's ok to drop support for older versions of openmc. We should update the documentation for moltres to reflect this requirement though.

**Comment by smpark7** [comment]
This is ready for another review @munkm 

I made the following changes:
- pep8 format edits to `moltres_xs.py`
- added checks whether OpenMC version is 0.13.2
- updated the readme to state that only OpenMC v0.13.2 and later are supported
- updated the pytest files which use the same files as the tutorial

**Review by munkm — APPROVED** [review]
This looks good to me! Thanks for clarifying that the .h5 files and the godiva.json were updated using the new openmc version.

**Comment by smpark7** [comment]
Thank you for reviewing this PR!

**Pull request merged**